components/brave_shields/browser/tracking_protection_service_browsertest.cc

```diff
@@ -34,105 +35,29 @@ const char kRedirectPage[] = "/client-redirect?";
-
 #if BUILDFLAG(BRAVE_STP_ENABLED)
   void SetUpDefaultCommandLine(base::CommandLine* command_line) override {
     InProcessBrowserTest::SetUpDefaultCommandLine(command_line);
     command_line->AppendSwitch(switches::kEnableSmartTrackingProtection);
   }
 #endif
 
-  void SetUp() override {
-    InitEmbeddedTestServer();
-    ExtensionBrowserTest::SetUp();
-  }
-
-  void SetUpOnMainThread() override {
-    ExtensionBrowserTest::SetUpOnMainThread();
-    base::PostTaskWithTraits(
-        FROM_HERE, {BrowserThread::IO},
-        base::BindOnce(&chrome_browser_net::SetUrlRequestMocksEnabled, true));
-    host_resolver()->AddRule("*", "127.0.0.1");
-  }
-
-  void PreRunTestOnMainThread() override {
-    ExtensionBrowserTest::PreRunTestOnMainThread();
-    WaitForTrackingProtectionServiceThread();
-    ASSERT_TRUE(
-        g_brave_browser_process->local_data_files_service()->IsInitialized());
-  }
-
-  void InitEmbeddedTestServer() {
-    brave::RegisterPathProvider();
-    base::FilePath test_data_dir;
-    base::PathService::Get(brave::DIR_TEST_DATA, &test_data_dir);
-    content::SetupCrossSiteRedirector(embedded_test_server());
-    embedded_test_server()->ServeFilesFromDirectory(test_data_dir);
-    ASSERT_TRUE(embedded_test_server()->Start());
-  }
-
-  void InitService() {
-    brave_component_updater::LocalDataFilesService::
-        SetComponentIdAndBase64PublicKeyForTest(
-            kTrackingProtectionComponentTestId,
-            kTrackingProtectionComponentTestBase64PublicKey);
-  }
-
-  void GetTestDataDir(base::FilePath* test_data_dir) {
-    base::ScopedAllowBlockingForTesting allow_blocking;
-    base::PathService::Get(brave::DIR_TEST_DATA, test_data_dir);
-  }
-
-  bool InstallTrackingProtectionExtension() {
-    base::FilePath test_data_dir;
-    GetTestDataDir(&test_data_dir);
-    const extensions::Extension* tracking_protection_extension =
-        InstallExtension(test_data_dir.AppendASCII("tracking-protection-data"),
-                         1);
-    if (!tracking_protection_extension)
-      return false;
-
-    g_brave_browser_process->tracking_protection_service()->OnComponentReady(
-        tracking_protection_extension->id(),
-        tracking_protection_extension->path(), "");
-    WaitForTrackingProtectionServiceThread();
-
-    return true;
-  }
-
-  void WaitForTrackingProtectionServiceThread() {
-    scoped_refptr<base::ThreadTestHelper> tr_helper(new base::ThreadTestHelper(
-        g_brave_browser_process->local_data_files_service()->GetTaskRunner()));
-    ASSERT_TRUE(tr_helper->Run());
-    scoped_refptr<base::ThreadTestHelper> io_helper(new base::ThreadTestHelper(
-        base::CreateSingleThreadTaskRunnerWithTraits({BrowserThread::IO})
-            .get()));
-    ASSERT_TRUE(io_helper->Run());
+  // BaseLocalDataFilesBrowserTest overrides
+  const char* test_data_directory() override { return kTestDataDirectory; }
+  const char* embedded_test_server_directory() override { return ""; }
```

Comment:
Any reason we're returning "" here whereas in the other override we're returning NULL? Seems like we should be consistent if possible.

Reply:
Better solution I think is to move the files used by the tracking protection browser test to their own subdirectory (they're not used by anything else), then let `embedded_test_server_directory()` return the non-empty path.